druid/src/widget/slider.rs

```diff
@@ -78,29 +127,477 @@ impl Slider {
         self
     }
 
-    /// check self.min <= self.max, if not swaps the values.
-    fn check_range(&mut self) {
-        if self.max < self.min {
-            warn!(
-                "min({}) should be less than max({}), swaping the values",
-                self.min, self.max
+    /// Builder-style method to set the track color.
+    ///
+    /// The default color is `None`.
+    pub fn track_color(mut self, color: impl Into<Option<KeyOrValue<Color>>>) -> Self {
+        self.track_color = color.into();
+        self
+    }
+
+    /// Builder-style method to set the knob style.
+    ///
+    /// The default is `Circle`.
+    pub fn knob_style(mut self, knob_style: KnobStyle) -> Self {
+        self.knob_style = knob_style;
+        self
+    }
+
+    /// Builder-style method to the the axis on which the slider moves.
+    ///
+    /// The default is `Horizontal`.
+    pub fn axis(mut self, axis: Axis) -> Self {
+        self.mapping.axis = axis;
+        self
+    }
+
+    /// Returns the Mapping of this Slider.
+    pub fn get_mapping(&self) -> SliderValueMapping {
+        self.mapping
+    }
+
+    /// Builder-style method to create an annotated range slider.
+    ///
+    pub fn annotated(self, named_steps: f64, unnamed_steps: f64) -> Annotated<f64, Self> {
```

Comment:
Giving either of these parameters an extremely low value or zero will stall the Slider example.

You should probably give these values a minimum and/or fire a warning if a value to low or negative is given.